batch/batch/driver/instance_pool.py

```diff
@@ -657,49 +666,52 @@ async def handle_call_delete_event(self, instance, timestamp):
         await instance.mark_deleted('deleted', timestamp)
 
     async def handle_event(self, event):
-        payload = event.get('jsonPayload')
+        payload = event.get('protoPayload')
         if payload is None:
             log.warning('event has no payload')
             return
 
         timestamp = dateutil.parser.isoparse(event['timestamp']).timestamp() * 1000
-        version = payload['version']
-        if version != '1.2':
-            log.warning('unknown event verison {version}')
-            return
 
         resource_type = event['resource']['type']
         if resource_type != 'gce_instance':
             log.warning(f'unknown event resource type {resource_type}')
             return
 
-        event_type = payload['event_type']
-        event_subtype = payload['event_subtype']
-        resource = payload['resource']
-        name = resource['name']
+        if 'last' in event['operation']:
+            event_type = 'GCE_OPERATION_DONE'
+        elif 'first' in event['operation']:
+            event_type = 'GCE_API_CALL'
+        else:
+            log.warning(f'unknown event_type {event["operation"]}')
+            event_type = None
+
+        event_subtype = payload['methodName']
+        resource = event['resource']
+        name = parse_resource_name(payload['resourceName'])['name']
 
-        log.info(f'event {version} {resource_type} {event_type} {event_subtype} {name}')
+        log.info(f'event {resource_type} {event_type} {event_subtype} {name}')
 
         if not name.startswith(self.machine_name_prefix):
             log.warning(f'event for unknown machine {name}')
             return
 
-        if event_subtype == 'compute.instances.insert':
+        if 'compute.instances.insert' in event_subtype:
             if event_type == 'GCE_OPERATION_DONE':
                 severity = event['severity']
-                operation_name = payload['operation']['name']
+                operation_type = 'insert'
                 success = (severity != 'ERROR')
-                self.zone_success_rate.push(resource['zone'], operation_name, success)
+                self.zone_success_rate.push(resource['labels']['zone'], operation_type, success)
         else:
             instance = self.name_instance.get(name)
             if not instance:
                 log.warning(f'event for unknown instance {name}')
                 return
 
-            if event_subtype == 'compute.instances.preempted':
+            if 'compute.instances.preempted' in event_subtype:
                 log.info(f'event handler: handle preempt {instance}')
                 await self.handle_preempt_event(instance, timestamp)
-            elif event_subtype == 'compute.instances.delete':
+            elif 'compute.instances.delete' in event_subtype:
```

Comment:
All these are just prefixed with `v1.`, right? I'm inclined to write out out the exact match, seems cleaner, less likely to cause an issue later, etc.